var/spack/repos/builtin/packages/libzip/package.py

```diff
@@ -46,14 +36,37 @@ def url_for_version(self, version):
 
     depends_on("zlib-api")
 
-    # Build system
-    build_system(
-        conditional("cmake", when="@1.4:"), conditional("autotools", when="@:1.3"), default="cmake"
-    )
+    variant("gnutls", default=True, description="Enable gnutls support")
+    variant("bzip2", default=True, description="Enable bzip2 support")
+    variant("lzma", default=True, description="Enable lzma support")
+    variant("openssl", default=True, description="Enable openssl support")
+    variant("zstd", default=True, description="Enable zstd support")
+    variant("mbedtls", default=True, description="Enable mbedtls support")
+    depends_on("gnutls", when="+gnutls")
+    depends_on("bzip2", when="+bzip2")
+    depends_on("lzma", when="+lzma")
+    depends_on("openssl", when="+openssl")
+    depends_on("mbedtls", when="+mbedtls")
+    # zstd support starts with version 1.8.0
+    with when("@1.8:"):
+        variant("zstd", default=True, description="Enable zstd support")
+        depends_on("zstd", when="+zstd")
 
     @property
     def headers(self):
         # Up to version 1.3.0 zipconf.h was installed outside of self.prefix.include
         return find_all_headers(
             self.prefix if self.spec.satisfies("@:1.3.0") else self.prefix.include
         )
+
+    def cmake_args(self):
+        return [
+            self.define_from_variant("ENABLE_GNUTLS", "gnutls"),
+            self.define_from_variant("ENABLE_MBEDTLS", "mbedtls"),
+            self.define_from_variant("ENABLE_OPENSSL", "openssl"),
+            self.define_from_variant("ENABLE_BZIP2", "bzip2"),
+            self.define_from_variant("ENABLE_LZMA", "lzma"),
+        ]
+        # Add zstd option if version is 1.8.0 or newer
+        if self.spec.satisfies("@1.8:"):
+            args.append(self.define_from_variant("ENABLE_ZSTD", "zstd"))
```

Comment:
I think this is better handled as a variant `zstd` when `@1.8:`, so there is no way to request `libzip@1.6.1 +zstd` which isn't possible.

Then this cmake_args clause can go unconditionally into the previous block since `self.define_from_variant("ENABLE_ZSTD", "zstd"),` will just work fine.

Reply:
done